osdk/tests/examples_in_book/test_and_run_projects.rs

```diff
@@ -15,25 +15,35 @@ fn create_and_run_kernel() {
         fs::remove_dir_all(&os_dir).unwrap();
     }
 
-    let mut command = cargo_osdk(&["new", "--kernel", os_name]);
-    command.current_dir(work_dir);
-    command.ok().unwrap();
+    let mut new_command = cargo_osdk(&["new", "--kernel", os_name]);
+    new_command.current_dir(work_dir);
+    new_command.ok().unwrap();
 
     // Makes the kernel depend on local OSTD
     let manifest_path = os_dir.join("Cargo.toml");
-    depends_on_local_ostd(&manifest_path);
-
-    let mut command = cargo_osdk(&["build"]);
-    command.current_dir(&os_dir);
-    command.ok().unwrap();
-
-    let mut command = cargo_osdk(&["run"]);
-    command.current_dir(&os_dir);
-    let output = command.output().unwrap();
-
+    depends_on_local_ostd(manifest_path);
+    let tdx_enabled = std::env::var("INTEL_TDX").is_ok();
+    if tdx_enabled {
+        let osdk_path = os_dir.join("OSDK.toml");
+        add_tdx_scheme_to_manifest(&osdk_path).unwrap();
```

Comment:
The TDX-related code are spread across the entire function, which is bad for readability. I suggest reorganizing the code this way (in pseudocode).

```
let opt_args = [];
let opt_config = [];
if tdx_support {
    opt_args += "--scheme tdx";
    opt_config += tdx_scheme;
}

cargo_osdk("build", opt_args);

cargo_osdk("run", opt_args);
```

This way all TDX-related code are placed in a central place and so they are easy to reason about.